support/cas-server-support-ehcache3-ticket-registry/src/main/java/org/apereo/cas/config/Ehcache3TicketRegistryConfiguration.java

```diff
@@ -0,0 +1,205 @@
+            false, managementAllAllEnabled, statisticsAllEnabled);
+        val configuration = new DefaultConfiguration(ehcacheProvider.getDefaultClassLoader(),
+            ehcache3CacheManagerConfiguration,
+            jsr107Config);
+        return ehcacheProvider.getCacheManager(ehcacheProvider.getDefaultURI(), configuration);
+    }
+
+    private CacheConfiguration<String, Ticket> buildCacheConfiguration(final TicketDefinition ticketDefinition) {
+        val ehcacheProperties = casProperties.getTicket().getRegistry().getEhcache3();
+        val terracottaClusterUri = ehcacheProperties.getTerracottaClusterUri();
+
+        CacheEventListenerConfigurationBuilder cacheEventListenerConfiguration = CacheEventListenerConfigurationBuilder
+            .newEventListenerConfiguration(new CasCacheEventListener(),
+                EventType.CREATED, EventType.UPDATED, EventType.EXPIRED, EventType.REMOVED, EventType.EVICTED)
+            .ordered().asynchronous();
+
+        val storageTimeout = ticketDefinition.getProperties().getStorageTimeout();
+        val expiryPolicy = ehcacheProperties.isEternal()
+            ?
+            ExpiryPolicyBuilder.noExpiration()
+            :
+            ExpiryPolicyBuilder.expiry()
+            .create(Duration.ofSeconds(storageTimeout))
+            .access(Duration.ofSeconds(storageTimeout))
+            .update(Duration.ofSeconds(storageTimeout)).build();
+
+        var resourcePools = ResourcePoolsBuilder.heap(ehcacheProperties.getMaxElementsInMemory());
+        if (StringUtils.isNotBlank(terracottaClusterUri)) {
+            resourcePools = resourcePools.with(
+                ClusteredResourcePoolBuilder.clusteredShared(ehcacheProperties.getResourcePoolName()));
+        }
+
+        resourcePools = resourcePools
+            .offheap(ehcacheProperties.getResourcePoolSize(), MemoryUnit.valueOf(ehcacheProperties.getResourcePoolUnits()));
+
+        if (StringUtils.isBlank(terracottaClusterUri)) {
+            resourcePools = resourcePools.disk(ehcacheProperties.getPerCacheSizeOnDisk(), MemoryUnit.valueOf(ehcacheProperties.getPerCacheSizeOnDiskUnits()));
+        }
+
+        var cacheConfigBuilder = CacheConfigurationBuilder.newCacheConfigurationBuilder(
+            String.class, Ticket.class, resourcePools)
+            .withExpiry(expiryPolicy)
+            .withService(cacheEventListenerConfiguration);
+
+        if (StringUtils.isNotBlank(terracottaClusterUri)) {
+            cacheConfigBuilder = cacheConfigBuilder.withService(
+                ClusteredStoreConfigurationBuilder.withConsistency(Consistency.STRONG));
+        }
+
+        return cacheConfigBuilder.build();
+    }
+
+    private static class CasCacheEventListener implements CacheEventListener<String, Ticket> {
+
+        @Override
+        public void onEvent(final CacheEvent<? extends String, ? extends Ticket> event) {
+            if (LOGGER.isTraceEnabled()) {
```

Comment:
The if condition here should be unnecessary unless `event.getXyz()` calls are expensive?